### ISSUE TITLE
Background process getting killed in 21301

### ISSUE THREAD

**Issue by syphernl** [owntracks/android#700]
With the recent build being in use for the past week I've noticed that the background process gets killed which wasn't the case with the previous/stable build. 

It happened about 3-4 times this week already.
 
I'm aware that OnePlus is rather aggressive in killing (background) apps but I haven't gotten an OTA for a few weeks now and am seeing different behavior, which happened after updating to build 21301.

* **App build number**: 21301
* **Android version**: 9
* **Device**: OnePlus 6T

**Comment by binarybucks** [comment]
Were you running 2.1.2 before that and have you checked the steps outlined in https://dontkillmyapp.com/oneplus to work around the restrictions? It could be that the app update reset those settings.

**Comment by syphernl** [comment]
After #695 I had to reinstall the app and had to re-disable the battery optimization for OwnTracks. I have just checked that this is still set to disabled.

**Comment by defcon84** [comment]
Since a few weeks, i have the same issue on a Galaxy S8 and S9.
Don't know if it crashes or gets optimized away.

**Comment by binarybucks** [comment]
Can you see if you can gather a debug log that contains the issue? 
There were quite some changes between versions but none that would explain why the service gets optimized away. Addtionally there were no reported crashes for the beta version yet.

**Comment by defcon84** [comment]
where can i find the debug log?

**Comment by binarybucks** [comment]
You can find instractions how to gather it here: https://owntracks.org/booklet/features/android/ 
Future versions will expose a GUI switch for it.

**Comment by defcon84** [comment]
Thanks, i've enabled it, will get back to you when i have something.

**Comment by syphernl** [comment]
For me it seems to happen most often when switching networks (e.g. 4G to WiFi)

**Comment by binarybucks** [comment]
Could be that my test WiFi connection tracking (that does nothing yet) crashes the service for some reason

**Comment by defcon84** [comment]
It finally happened again. :-)
Today 22-03-2019 around 8:30 i noticed it wasn't running anymore.
I'll add the last of the logs, hopefully it makes sense to you.
I don't see anything weird.

```
21-03-2019.html last bit
  do mrt. 21 2019 at 07:39:22:035 p.m. :    location update received: tst:1553193561910, acc:16.023, lat:1.354885, lon:1.4567601
  do mrt. 21 2019 at 07:39:22:041 p.m. :    trigger:null
  do mrt. 21 2019 at 07:39:22:058 p.m. :    messageId:1553193562045, queueLength:1, queue:{1553193562045=org.owntracks.android.messages.MessageLocation@a7abef6}
  do mrt. 21 2019 at 07:39:22:064 p.m. :    getting first message from queue: 1553193562045
  do mrt. 21 2019 at 07:39:22:073 p.m. :    Thread: Thread[pool-5-thread-1,5,main]
  do mrt. 21 2019 at 07:39:22:088 p.m. :    already connected
  do mrt. 21 2019 at 07:39:22:183 p.m. :    message sent: 1553193562045
  do mrt. 21 2019 at 07:39:22:188 p.m. :    processing location message owntracks/mymqttuser/user1
  do mrt. 21 2019 at 07:39:22:194 p.m. :    messageId:1553193562045, queueLength:1
  do mrt. 21 2019 at 07:39:22:197 p.m. :    update contact:owntracks/mymqttuser/user1, tst:1553193561
  do mrt. 21 2019 at 07:39:22:204 p.m. :    MessageLocation received org.owntracks.android.messages.MessageLocation@a7abef6, org.owntracks.android.messages.MessageLocation@b6c2a2a, outgoing: false
  do mrt. 21 2019 at 07:39:22:204 p.m. :    processing location message owntracks/mymqttuser/user1
  do mrt. 21 2019 at 07:39:22:215 p.m. :    cache lookup for 1553193562045 (hash 1.354885-1.456760) -> null
  do mrt. 21 2019 at 07:39:22:222 p.m. :    messageId:1553193562045
  do mrt. 21 2019 at 07:39:22:226 p.m. :    queue empty
  do mrt. 21 2019 at 07:39:22:374 p.m. :    geocoding result: ******
  do mrt. 21 2019 at 07:41:05:685 p.m. :    SendLocationPingWorker doing work
  do mrt. 21 2019 at 07:41:05:693 p.m. :    trigger:p
  do mrt. 21 2019 at 07:41:05:723 p.m. :    messageId:1553193665705, queueLength:1, queue:{1553193665705=org.owntracks.android.messages.MessageLocation@d554793}
  do mrt. 21 2019 at 07:41:05:728 p.m. :    getting first message from queue: 1553193665705
  do mrt. 21 2019 at 07:41:05:748 p.m. :    Thread: Thread[pool-5-thread-2,5,main]
  do mrt. 21 2019 at 07:41:05:757 p.m. :    already connected
  do mrt. 21 2019 at 07:41:05:782 p.m. :    message sent: 1553193665705
  do mrt. 21 2019 at 07:41:05:789 p.m. :    messageId:1553193665705, queueLength:1
  do mrt. 21 2019 at 07:41:05:797 p.m. :    MessageLocation received org.owntracks.android.messages.MessageLocation@d554793, org.owntracks.android.messages.MessageLocation@a7abef6, outgoing: false
  do mrt. 21 2019 at 07:41:05:805 p.m. :    processing location message owntracks/mymqttuser/user1
  do mrt. 21 2019 at 07:41:05:818 p.m. :    cache lookup for 1553193665705 (hash 1.354885-1.456760) -> ******
  do mrt. 21 2019 at 07:41:05:822 p.m. :    update contact:owntracks/mymqttuser/user1, tst:1553193665
  do mrt. 21 2019 at 07:41:05:831 p.m. :    messageId:1553193665705
  do mrt. 21 2019 at 07:41:05:834 p.m. :    processing location message owntracks/mymqttuser/user1
  do mrt. 21 2019 at 07:41:05:838 p.m. :    queue empty
  do mrt. 21 2019 at 08:36:56:603 p.m. :    message:null,
  do mrt. 21 2019 at 08:36:56:627 p.m. :    instantiating new outgoingMessageProcessorExecutor
  do mrt. 21 2019 at 08:36:56:653 p.m. :    WorkManager queue task PERIODIC_TASK_MQTT_RECONNECT as e23dcbec-87ef-4359-b5ac-1076c64201ce

22-03-2019.html
  vr mrt. 22 2019 at 05:52:29:624 a.m. :    message:null,
  vr mrt. 22 2019 at 05:52:29:649 a.m. :    instantiating new outgoingMessageProcessorExecutor
  vr mrt. 22 2019 at 05:52:29:663 a.m. :    WorkManager queue task PERIODIC_TASK_MQTT_RECONNECT as b15ce0c6-757b-4bfe-b4bc-a34de5cf541d
```

**Comment by binarybucks** [comment]
Indeed, there seems to be nothing wrong. 
Can you give 21302 a try? I disabled the network detection code that could cause this.

**Comment by defcon84** [comment]
21201 is latest in the play store

**Comment by binarybucks** [comment]
Ah sorry, I meant the 21302 beta build

**Comment by defcon84** [comment]
I've never done something like that :)
Do i need an apk file that i'll copy to my phone?

**Comment by binarybucks** [comment]
It's even easier. Just register at https://play.google.com/apps/testing/org.owntracks.android and you will receive the beta build as a normal update through Google Play.

**Comment by defcon84** [comment]
That was indeed easy!
I have the new version running now.
Now I wait again.

**Comment by defcon84** [comment]
4 days in, still hasn't stopped.

**Comment by binarybucks** [comment]
Neat, thank you for the feedback

**Comment by syphernl** [comment]
I can confirm the same, with 21302 the background process hasn't crashed and has been up for a week. Problem resolved! :+1:

**Comment by defcon84** [comment]
I regret to inform you that it happened again for me.
Do i need to create a new issue?
If not, here is the log :)
```
 wo mrt. 27 2019 at 10:21:03:841 a.m./2/App :    Planted trees :org.owntracks.android.support.TimberLogFileTree@de5f248
  wo mrt. 27 2019 at 10:21:03:878 a.m./4/MQTTReconnectWorker :    MQTTReconnectWorker Doing work (Thread[pool-1-thread-1,5,main])
  wo mrt. 27 2019 at 10:21:04:356 a.m./2/MessageProcessor :    message:null,
  wo mrt. 27 2019 at 10:21:04:360 a.m./2/MessageProcessor :    instantiating new outgoingMessageProcessorExecutor
  wo mrt. 27 2019 at 10:21:04:366 a.m./2/Scheduler :    WorkManager queue task PERIODIC_TASK_MQTT_RECONNECT as 99d80d69-36f7-4cd2-83a9-d8f06633190a
  wo mrt. 27 2019 at 10:21:04:494 a.m./4/MQTTReconnectWorker :    MQTTReconnectWorker Doing work (Thread[pool-1-thread-2,5,main])
  wo mrt. 27 2019 at 10:21:04:496 a.m./6/MessageProcessorEndpointMqtt :    Thread: Thread[pool-1-thread-2,5,main]
  wo mrt. 27 2019 at 10:21:04:497 a.m./2/MessageProcessorEndpointMqtt :    connecting on thread 34686
  wo mrt. 27 2019 at 10:21:04:499 a.m./2/MessageProcessor :    message:null,
  wo mrt. 27 2019 at 10:21:04:501 a.m./2/MessageProcessorEndpointMqtt :    initializing new mqttClient
  wo mrt. 27 2019 at 10:21:04:504 a.m./2/MessageProcessorEndpointMqtt :    mode: 0
  wo mrt. 27 2019 at 10:21:04:506 a.m./2/MessageProcessorEndpointMqtt :    client id: mqtt-useruser1
  wo mrt. 27 2019 at 10:21:04:507 a.m./2/MessageProcessorEndpointMqtt :    connect string: ssl://mydomain.com:8883
  wo mrt. 27 2019 at 10:21:04:592 a.m./2/SocketFactory :    CA sideload: false, using system keystore
  wo mrt. 27 2019 at 10:21:04:599 a.m./2/MessageProcessorEndpointMqtt :    connecting sync
  wo mrt. 27 2019 at 10:21:05:983 a.m./2/MessageProcessorEndpointMqtt :    false, serverUri:ssl://mydomain.com:8883
  wo mrt. 27 2019 at 10:21:05:983 a.m./2/Scheduler :    WorkManager queue task PERIODIC_TASK_MQTT_KEEPALIVE as 2c641a81-71a9-466f-a124-a22eea8e1c21 with interval 3600
  wo mrt. 27 2019 at 10:21:05:984 a.m./2/Scheduler :    Cancelling task tag PERIODIC_TASK_MQTT_RECONNECT
  wo mrt. 27 2019 at 10:21:05:986 a.m./2/MessageProcessorEndpointMqtt :    subscribe() - Will subscribe to: owntracks/+/+
  wo mrt. 27 2019 at 10:21:05:988 a.m./2/MessageProcessorEndpointMqtt :    subscribe() - Will subscribe to: owntracks/+/+/info
  wo mrt. 27 2019 at 10:21:05:990 a.m./2/MessageProcessorEndpointMqtt :    subscribe() - Will subscribe to: owntracks/mqtt-user/user1/cmd
  wo mrt. 27 2019 at 10:21:05:991 a.m./2/MessageProcessor :    message:null,
  wo mrt. 27 2019 at 10:21:05:992 a.m./2/MessageProcessorEndpointMqtt :    subscribe() - Will subscribe to: owntracks/+/+/event
  wo mrt. 27 2019 at 10:21:05:994 a.m./2/MessageProcessorEndpointMqtt :    subscribe() - Will subscribe to: owntracks/+/+/waypoint
  wo mrt. 27 2019 at 10:21:06:112 a.m./2/MQTTKeepaliveWorker :    MQTTKeepaliveWorker doing work
  wo mrt. 27 2019 at 10:21:06:193 a.m./2/MessageProcessor :    type:unknown, key:owntracks/mqtt-user/user1a
  wo mrt. 27 2019 at 10:21:06:195 a.m./2/MessageProcessor :    type:unknown, key:owntracks/mqtt-user/user2a
  wo mrt. 27 2019 at 10:21:06:236 a.m./6/MessageProcessorEndpointMqtt :    connectionLost error Connection lost (32109) - javax.net.ssl.SSLException: Read error: ssl=0x74eb071488: I/O error during system call, Software caused connection abort at org.eclipse.paho.client.mqttv3.internal.CommsReceiver.run(CommsReceiver.java:181) at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:459) at java.util.concurrent.FutureTask.run(FutureTask.java:266) at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301) at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167) at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641) at java.lang.Thread.run(Thread.java:764) Caused by: javax.net.ssl.SSLException: Read error: ssl=0x74eb071488: I/O error during system call, Software caused connection abort at com.android.org.conscrypt.NativeCrypto.SSL_read(Native Method) at com.android.org.conscrypt.NativeSsl.read(NativeSsl.java:407) at com.android.org.conscrypt.ConscryptFileDescriptorSocket$SSLInputStream.read(ConscryptFileDescriptorSocket.java:579) at com.android.org.conscrypt.ConscryptFileDescriptorSocket$SSLInputStream.read(ConscryptFileDescriptorSocket.java:550) at java.io.DataInputStream.readByte(DataInputStream.java:268) at org.eclipse.paho.client.mqttv3.internal.wire.MqttInputStream.readMqttWireMessage(MqttInputStream.java:92) at org.eclipse.paho.client.mqttv3.internal.CommsReceiver.run(CommsReceiver.java:133) ... 6 more
  wo mrt. 27 2019 at 10:21:06:238 a.m./2/Scheduler :    Cancelling task tag PERIODIC_TASK_MQTT_KEEPALIVE
  wo mrt. 27 2019 at 10:21:06:254 a.m./2/Scheduler :    WorkManager queue task PERIODIC_TASK_MQTT_RECONNECT as 07c07a42-44a6-4066-9681-31b67d03c732
  wo mrt. 27 2019 at 10:21:06:257 a.m./6/MessageProcessorEndpointMqtt :    message: Connection lost java.lang.Exception: Connection lost (32109) - javax.net.ssl.SSLException: Read error: ssl=0x74eb071488: I/O error during system call, Software caused connection abort at org.owntracks.android.services.MessageProcessorEndpointMqtt$1.connectionLost(MessageProcessorEndpointMqtt.java:127) at org.eclipse.paho.client.mqttv3.internal.CommsCallback.connectionLost(CommsCallback.java:292) at org.eclipse.paho.client.mqttv3.internal.ClientComms.shutdownConnection(ClientComms.java:423) at org.eclipse.paho.client.mqttv3.internal.CommsReceiver.run(CommsReceiver.java:181) at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:459) at java.util.concurrent.FutureTask.run(FutureTask.java:266) at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301) at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167) at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641) at java.lang.Thread.run(Thread.java:764) Caused by: Connection lost (32109) - javax.net.ssl.SSLException: Read error: ssl=0x74eb071488: I/O error during system call, Software caused connection abort ... 7 more Caused by: javax.net.ssl.SSLException: Read error: ssl=0x74eb071488: I/O error during system call, Software caused connection abort at com.android.org.conscrypt.NativeCrypto.SSL_read(Native Method) at com.android.org.conscrypt.NativeSsl.read(NativeSsl.java:407) at com.android.org.conscrypt.ConscryptFileDescriptorSocket$SSLInputStream.read(ConscryptFileDescriptorSocket.java:579) at com.android.org.conscrypt.ConscryptFileDescriptorSocket$SSLInputStream.read(ConscryptFileDescriptorSocket.java:550) at java.io.DataInputStream.readByte(DataInputStream.java:268) at org.eclipse.paho.client.mqttv3.internal.wire.MqttInputStream.readMqttWireMessage(MqttInputStream.java:92) at org.eclipse.paho.client.mqttv3.internal.CommsReceiver.run(CommsReceiver.java:133) ... 6 more
  wo mrt. 27 2019 at 10:21:06:259 a.m./2/MessageProcessor :    message:Connection lost,
  wo mrt. 27 2019 at 10:21:06:309 a.m./2/MessageProcessor :    processing location message owntracks/mqtt-user/user1
  wo mrt. 27 2019 at 10:21:06:311 a.m./2/FusedContact :    update contact:owntracks/mqtt-user/user1, tst:1553642092
  wo mrt. 27 2019 at 10:21:06:313 a.m./2/MemoryContactsRepo :    new contact allocated id:owntracks/mqtt-user/user1, tid:fc
  wo mrt. 27 2019 at 10:27:51:299 a.m./2/SendLocationPingWorker :    SendLocationPingWorker doing work
  wo mrt. 27 2019 at 10:27:51:305 a.m./2/LocationProcessor :    trigger:p
  wo mrt. 27 2019 at 10:27:51:311 a.m./4/MQTTReconnectWorker :    MQTTReconnectWorker Doing work (Thread[pool-1-thread-1,5,main])
  wo mrt. 27 2019 at 10:27:51:313 a.m./6/MessageProcessorEndpointMqtt :    Thread: Thread[pool-1-thread-1,5,main]
  wo mrt. 27 2019 at 10:27:51:314 a.m./2/MessageProcessorEndpointMqtt :    connecting on thread 34683
  wo mrt. 27 2019 at 10:27:51:316 a.m./2/MessageProcessor :    message:null,
  wo mrt. 27 2019 at 10:27:51:319 a.m./2/SocketFactory :    CA sideload: false, using system keystore
  wo mrt. 27 2019 at 10:27:51:323 a.m./6/LocationProcessor :    no location available
  wo mrt. 27 2019 at 10:27:51:342 a.m./2/MessageProcessorEndpointMqtt :    connecting sync
  wo mrt. 27 2019 at 10:29:24:429 a.m./6/MessageProcessorEndpointMqtt :    message: Connection lost Connection lost (32109) - java.io.EOFException at org.eclipse.paho.client.mqttv3.internal.CommsReceiver.run(CommsReceiver.java:181) at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:459) at java.util.concurrent.FutureTask.run(FutureTask.java:266) at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301) at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167) at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641) at java.lang.Thread.run(Thread.java:764) Caused by: java.io.EOFException at java.io.DataInputStream.readByte(DataInputStream.java:270) at org.eclipse.paho.client.mqttv3.internal.wire.MqttInputStream.readMqttWireMessage(MqttInputStream.java:92) at org.eclipse.paho.client.mqttv3.internal.CommsReceiver.run(CommsReceiver.java:133) ... 6 more
  wo mrt. 27 2019 at 10:29:24:436 a.m./2/MessageProcessor :    message:Connection lost,
  wo mrt. 27 2019 at 01:44:26:775 p.m./2/App :    Planted trees :org.owntracks.android.support.TimberLogFileTree@c7bc606
  wo mrt. 27 2019 at 01:44:26:975 p.m./2/SendLocationPingWorker :    SendLocationPingWorker doing work
  wo mrt. 27 2019 at 01:44:26:976 p.m./2/LocationProcessor :    trigger:p
  wo mrt. 27 2019 at 01:44:26:979 p.m./6/LocationProcessor :    no location available
  wo mrt. 27 2019 at 01:44:27:046 p.m./4/MQTTReconnectWorker :    MQTTReconnectWorker Doing work (Thread[pool-1-thread-2,5,main])
  wo mrt. 27 2019 at 01:44:27:288 p.m./2/MessageProcessor :    message:null,
  wo mrt. 27 2019 at 01:44:27:289 p.m./2/MessageProcessor :    instantiating new outgoingMessageProcessorExecutor
  wo mrt. 27 2019 at 01:44:27:292 p.m./2/Scheduler :    WorkManager queue task PERIODIC_TASK_MQTT_RECONNECT as ebe29d5b-2a00-4671-9ad9-889b0ee6085d
  wo mrt. 27 2019 at 01:44:30:379 p.m./2/SendLocationPingWorker :    SendLocationPingWorker doing work
  wo mrt. 27 2019 at 01:44:30:381 p.m./2/LocationProcessor :    trigger:p
  wo mrt. 27 2019 at 01:44:30:426 p.m./6/LocationProcessor :    no location available
  wo mrt. 27 2019 at 01:44:30:679 p.m./4/MQTTReconnectWorker :    MQTTReconnectWorker Doing work (Thread[pool-1-thread-4,5,main])
  wo mrt. 27 2019 at 01:44:30:681 p.m./6/MessageProcessorEndpointMqtt :    Thread: Thread[pool-1-thread-4,5,main]
  wo mrt. 27 2019 at 01:44:30:682 p.m./2/MessageProcessorEndpointMqtt :    connecting on thread 35344
  wo mrt. 27 2019 at 01:44:30:686 p.m./2/MessageProcessor :    message:null,
  wo mrt. 27 2019 at 01:44:30:687 p.m./2/MessageProcessorEndpointMqtt :    initializing new mqttClient
  wo mrt. 27 2019 at 01:44:30:690 p.m./2/MessageProcessorEndpointMqtt :    mode: 0
  wo mrt. 27 2019 at 01:44:30:777 p.m./2/MessageProcessorEndpointMqtt :    client id: mqtt-useruser1
  wo mrt. 27 2019 at 01:44:30:779 p.m./2/MessageProcessorEndpointMqtt :    connect string: ssl://mydomain.com:8883
  wo mrt. 27 2019 at 01:44:31:223 p.m./2/SocketFactory :    CA sideload: false, using system keystore
  wo mrt. 27 2019 at 01:44:31:300 p.m./2/MessageProcessorEndpointMqtt :    connecting sync
  wo mrt. 27 2019 at 01:44:35:067 p.m./2/Scheduler :    WorkManager queue task PERIODIC_TASK_MQTT_KEEPALIVE as a5728ee8-c25c-4adc-bdee-036f91ec26ea with interval 3600
  wo mrt. 27 2019 at 01:44:35:069 p.m./2/MessageProcessorEndpointMqtt :    false, serverUri:ssl://mydomain.com:8883
  wo mrt. 27 2019 at 01:44:35:072 p.m./2/Scheduler :    Cancelling task tag PERIODIC_TASK_MQTT_RECONNECT
  wo mrt. 27 2019 at 01:44:35:073 p.m./2/MessageProcessor :    message:null,
  wo mrt. 27 2019 at 01:44:35:086 p.m./2/MessageProcessorEndpointMqtt :    subscribe() - Will subscribe to: owntracks/+/+
  wo mrt. 27 2019 at 01:44:35:088 p.m./2/MessageProcessorEndpointMqtt :    subscribe() - Will subscribe to: owntracks/+/+/info
  wo mrt. 27 2019 at 01:44:35:090 p.m./2/MessageProcessorEndpointMqtt :    subscribe() - Will subscribe to: owntracks/mqtt-user/user1/cmd
  wo mrt. 27 2019 at 01:44:35:109 p.m./2/MessageProcessorEndpointMqtt :    subscribe() - Will subscribe to: owntracks/+/+/event
  wo mrt. 27 2019 at 01:44:35:111 p.m./2/MessageProcessorEndpointMqtt :    subscribe() - Will subscribe to: owntracks/+/+/waypoint
  wo mrt. 27 2019 at 01:44:35:358 p.m./2/MQTTKeepaliveWorker :    MQTTKeepaliveWorker doing work
  wo mrt. 27 2019 at 01:44:35:961 p.m./2/MessageProcessor :    processing location message owntracks/mqtt-user/user1
  wo mrt. 27 2019 at 01:44:35:962 p.m./2/FusedContact :    update contact:owntracks/mqtt-user/user1, tst:1553642092
  wo mrt. 27 2019 at 01:44:35:964 p.m./2/MemoryContactsRepo :    new contact allocated id:owntracks/mqtt-user/user1, tid:fc
  wo mrt. 27 2019 at 01:44:35:966 p.m./2/MessageProcessor :    processing location message owntracks/mqtt-user/user2
  wo mrt. 27 2019 at 01:44:35:967 p.m./2/FusedContact :    update contact:owntracks/mqtt-user/user2, tst:1553586778
  wo mrt. 27 2019 at 01:44:35:969 p.m./2/MemoryContactsRepo :    new contact allocated id:owntracks/mqtt-user/user2, tid:lb
  wo mrt. 27 2019 at 01:44:36:020 p.m./2/MessageProcessor :    type:base, key:owntracks/mqtt-user/user1
  wo mrt. 27 2019 at 01:44:36:035 p.m./2/MessageProcessor :    type:unknown, key:owntracks/mqtt-user/user1a
  wo mrt. 27 2019 at 01:44:36:038 p.m./2/MessageProcessor :    type:unknown, key:owntracks/mqtt-user/user2a
  wo mrt. 27 2019 at 01:44:36:040 p.m./2/MessageProcessor :    processing location message owntracks/mqtt-user/user1
  wo mrt. 27 2019 at 01:44:36:043 p.m./2/MessageProcessor :    processing location message owntracks/mqtt-user/user2
  wo mrt. 27 2019 at 01:44:36:045 p.m./2/MessageProcessor :    type:unknown, key:owntracks/mqtt-user/user1a/info
  wo mrt. 27 2019 at 01:44:36:048 p.m./2/MessageProcessor :    type:unknown, key:owntracks/mqtt-user/user2a/info
```

**Comment by mf-social** [comment]
Just a quick note to say this is happening to me too on a pixel 2. I haven't looked into the logging and stuff yet as I only just started to investigate the issue, but randomly the service stops and I have to go in to the app to restart it.

If I come up with anything useful for debugging I'll post further.

**Comment by defcon84** [comment]
It just keeps on happening almost exactly every 4 days. Probably some Android 9 thingy that optimizes the app away.

**Comment by binarybucks** [comment]
I assume you have Settings > Apps > Owntracks > Advanced > Battery > Background restrictions set to correct values?

**Comment by defcon84** [comment]
As far as I can see it is unrestricted

**Comment by kirichkov** [comment]
This sounds like the issue I have, so I'll add the information I have - I'm seeing the same behavior on two phones - Samsung J1 2016 (Running Android 5.x AFAIK) and a Xiaomi Redmi Note 4 (Android 8, I think). Both of these phones have the battery optimization turned off and OT has been working for at least a year on both phones without any hiccups. Yesterday I noticed that around 27th of March they both stopped reporting. I've restarted owntracks, did a test drive, but nothing was reported until I returned to the Wi-Fi area. I've restarted the phone, disconnected from the Wi-Fi and now it seems to be reporting properly.

The version used is the one from the App Store, I'm mentioning the date, since it might be related to a new version that's been pushed to the app store and I updated to it. Given that the J1 hasn't had any firmware updates for ages now, I don't think this is related to Android.

**Comment by defcon84** [comment]
I have finally found my issue.
It seems that Samsung has added a setting that says "Put unused apps to sleep".
The phone thinks I'm not using Owntracks so it closes it after 3 days.
https://imgur.com/HUHjFXY
I have disabled this setting and now it keeps on working.

Is there a way that you can let Android know that the app is still in use?

**Comment by growse** [comment]
> Is there a way that you can let Android know that the app is still in use?

I suspect that in this case, "use" means "in the foreground". Also, this seems like it's a Samsung-specific feature, rather than something available in the Android SDK?

**Comment by binarybucks** [comment]
Owntracks behaves to the spec, Samsung just chooses to ignore it. Hence there will be no change from our side.

I suggest to add the workaround to our booklet.

**Comment by defcon84** [comment]
> Also, this seems like it's a Samsung-specific feature, rather than something available in the Android SDK?
My guess is that it is Samsung specific

> Owntracks behaves to the spec, Samsung just chooses to ignore it. Hence there will be no change from our side.
> 
> I suggest to add the workaround to our booklet.

I can live with that.
